src/App.tsx

```diff
@@ -1,9 +1,15 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+
+  const handleChange: React.ChangeEventHandler<HTMLInputElement> = (event) => {
+    setQuery(event.target.value);
+  };
+
```

Comment:
you should add logic here, create variable e.g. `const visibleMovies` which will be passed as props `movies={visibleMovies}` to your `MoviesList` component, you should use the filter method to handle this logic based on what you stored in `query` state, 

```
const visibleMovies = moviesFromServer
    .filter(etc..... logic here)
```